### PR TITLE
Update .travis.yml

### PR DESCRIPTION
**Pull request by OutKa5t**
Changed deployment user to be app specific.
Created secret vars and shuffled things around so its clear whats what

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -25,45 +25,50 @@ env:
...
+  script: bash deploy.sh staging
   on: staging
 - provider: script
-  script: git fetch --unshallow && git push --force https://govuk-dfe-bat-deployment-user:$AZURE_WA_PASSWORD@bat-prod-manage-courses-api-app.scm.azurewebsites.net:443/bat-prod-manage-courses-api-app.git
-    HEAD:refs/heads/master
+  script: bash deploy.sh prod
   on: production
 - provider: script
-  script: "./travis-deploy-sc-exporter.sh govuk-dfe-bat-deployment-user:$AZURE_WA_PASSWORD
+  script: "./travis-deploy-sc-exporter.sh $bat-dev-manage-courses-api-app:$APP_CREDENTIALS_DEV_PWD
```
Comment (misaka) [inline .travis.yml]: `$bat` needs escaping here -> `\$bat`

Comment (misaka) [reply .travis.yml]: Here and all the instances of `$bat` below.

File: deploy.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+set -e
+
+# script for deploying to any environment, choosing the matching deploy password environment variable
+
+# arg 1: environment to deploy to
+env=$1
+
+# convert env to uppercase to match environment variable case
+upperenv=`echo "$env" | tr '[:lower:]' '[:upper:]'
+`
```
Comment (misaka) [inline deploy.sh]: Stray newline/` here.

File: .travis.yml (hunk 1 above)
Comment (tvararu) [reply .travis.yml]: Yes, it did need it

**Comment by timabell** [comment]
[Fix deployment for all Azure apps using local git](https://trello.com/c/PUuF1Gzt/896-fix-deployment-for-all-azure-apps-using-local-git)

**Pull request merged**